packages/app/src/pages/popup/send.tsx

```diff
@@ -24,83 +25,102 @@ interface FormValues {
+    }
+    if (!validateStacksAddress(recipient)) {
+      errors.recipient = 'The address you provided is not valid.';
+    }
+    if (recipient === currentAccountStxAddress) {
+      errors.recipient = 'Cannot send to yourself.';
+    }
+    if (amount === '') {
+      errors.amount = 'You must specify an amount.';
+    }
+    if (amount <= 0) {
+      errors.amount = 'Must be more than zero.';
+    }
+    if (selectedAsset) {
+      if (balances.value) {
+        const amountBN = new BigNumber(amount);
+        if (selectedAsset.type === 'stx') {
+          const curBalance = microStxToStx(balances.value.stx.balance);
+          if (curBalance.lt(amountBN)) {
+            errors.amount = `You don't have enough tokens. Your balance is ${curBalance.toString()}`;
+          }
+        }
+      }
+      if (selectedAsset.type === 'ft') {
+        const { address, contractName } = getAssetStringParts(selectedAsset.contractAddress);
+        if (amount.toString().includes('.')) {
+          errors.amount = 'When sending a fungible token, amount must be an integer.';
+        }
+        try {
+          const rpcClient = new RPCClient(currentNetwork.url);
+          const contractInterface = await rpcClient.fetchContractInterface({
+            contractAddress: address,
+            contractName,
+          });
+          const transferFunction = contractInterface.functions.find(func => {
+            const correctName = func.name === 'transfer';
+            const [recipientArg, amountArg] = func.args;
+            const correctRecipient =
+              recipientArg?.name === 'recipient' && recipientArg?.type === 'principal';
+            const correctAmount = amountArg?.name === 'amount' && amountArg?.type === 'uint128';
+            return correctName && correctRecipient && correctAmount && func.args.length === 2;
+          });
+          if (!transferFunction) {
+            setAssetError('The contract you specified does not have a `transfer` function.');
+          }
+        } catch (error) {
+          console.error(error);
+          setAssetError('Unable to fetch contract details.');
+        }
+      }
+    } else {
+      setAssetError('You must select a valid token to transfer.');
+    }
+    return errors;
+  };
+};
+
 export const PopupSend: React.FC = () => {
   const { doChangeScreen } = useAnalytics();
-  const { currentNetwork, currentAccountStxAddress } = useWallet();
   const [isShowing, setShowing] = useState(false);
   const [assetError, setAssetError] = useState<string | undefined>();
   const balances = useFetchBalances();
   const selectedAsset = useRecoilValue(selectedAssetStore);
+  const assets = useAssets();
+  const amountInputRef = useRef<HTMLInputElement | null>(null);
   const initialValues: FormValues = {
     amount: '',
     recipient: '',
   };
-  return (
+
+  useEffect(() => {
+    if (assets.length === 1 && amountInputRef.current) {
+      amountInputRef?.current?.focus?.();
+    }
+  }, [amountInputRef]);
```

Comment:
We should add `assets.length` as a dep here, or is there a good reason why not to?

Gotta handle that edge case where the user receives an asset while they're on the send page, ya know 😉?